src/workload/mod.rs

```diff
@@ -510,6 +555,48 @@ impl Distribution {
 
 impl Keyspace {
     pub fn new(config: &Config, keyspace: &config::Keyspace) -> Self {
+        let value_random_bytes = if keyspace.compression_ratio() <= 1.0 || keyspace.vlen().is_none()
+        {
+            // this indicates the message should not be compressible, to achieve
```

Comment:
I think it is worth hoisting this block into a separate static structure so that both Keyspace and Topics can use it. Also, if we do shuffle the message before sending, we should probably do that here too.

Reply:
Factored out.